### DISCUSSION TITLE
Formalize Preemption Hierarchy

### DISCUSSION THREAD

**Discussion by katelynsills** [Agoric/agoric-sdk#11226] [Ideas]
The Pixel Demo was our first exploration of a hierarchy of rights in which holders of the more fundamental property right can evict the temporary users of the right, similar to how owners/tenants/subtenants interact. We need to formalize this as a feature.

**Comment by DavidBruant** [comment]
I would be interested in participating in this

The next documentation guide i was imagining is a mint producing hierarchical rights like the Pixel Demo does
I've been reading the code around the Pixel Demo for this and i have to admit that it's been too had for me to understand fully the intent and why the existing Pixel Demo code is the only/easiest way of doing it

**Comment by katelynsills** [comment]
Ok, that sounds good. I think there's a step before a new hierarchical rights example, which would be a non-hierarchical example that has use rights, like the stock example which has the right to vote and right to dividends #73. 

I think after #385, the pixel demo will probably need to be rewritten. For example, if payments are not meant to be trusted ever, we shouldn't be adding methods to them to get use objects. We also can't get the assay or parent/childAssay from the payment.
